openedx/features/edly/utils.py

```diff
@@ -0,0 +1,255 @@
+            absolute_urls.append(absolute_url)
+        else:
+            # This url doesn't belong to the course we are not changing it.
+            relative_urls.remove(relative_url)
+    absolute_urls = ["{lms_base_url}{path}".format(lms_base_url=settings.LMS_ROOT_URL, path=path)
+                     for path in absolute_urls]
+    return relative_urls, absolute_urls
+
+
+def _is_static_path(path):
+    return path.startswith('/static/')
+
+
+def _is_canonicalized_asset_path(path):
+    return path.startswith('/asset')
+
+
+def _generate_absolute_url_from_static_path(course_key, path):
+    """
+    This function will generate the exact path of the given static path.
+
+    :param course_key: Course for which we have to create the static assets path.
+    :param path: Static path of asset.
+    :return: Absolute path of the static asset.
+    """
+    absolute_url = StaticContent.get_canonicalized_asset_path(course_key, path, "", {})
+    return absolute_url
+
+
+def get_xblock(usage_key):
+    """
+    This function will get xblock.
+
+    :param usage_key: usage_key of xblock we are getting.
+    :return: xblock
+    """
+    store = modulestore()
+    with store.bulk_operations(usage_key.course_key):
+        try:
+            return store.get_item(usage_key, depth=None)
+        except ItemNotFoundError:
+            log.error("Can't find parent xblock.")
+            return None
+        except InvalidLocationError:
+            log.error("Can't find parent/item by location.")
+            return None
+
+
+def _get_published_unit_url(xblock):
+    """
+    This function will generate the url of unit published
+
+    :param xblock: Unit xblock published
+    :return: Learner site url of unit.
+
+    URL pattern:  LMS_ROOT_URL/courses/courseware/SECTION_URL_NAME/SUBSECTION_URL_NAME/UNIT_URL_NAME/TAB_NUM?
+                    activate_block_id=UNIT_XBLOCK_USAGE_KEY
+    """
+    published_unit_url = None
+    if xblock.parent:
+        subsection_xblock = get_xblock(xblock.parent)
+        if subsection_xblock:
+            section_xblock = get_xblock(subsection_xblock.parent)
+            if section_xblock:
+                subsection_url_name = subsection_xblock.url_name
+                section_url_name = section_xblock.url_name
+                published_unit_url = '{lms_base_url}/courses/{course_id}/courseware/{section_url_name}/' \
+                                     '{subsection_url_name}/?activate_block_id={unit_usage_key}'.format(
+                    lms_base_url=settings.LMS_ROOT_URL,
+                    course_id=xblock.location.course_key,
+                    section_url_name=section_url_name,
+                    subsection_url_name=subsection_url_name,
+                    unit_usage_key=xblock.location)
+    return published_unit_url
+
+
+def get_course_enrollments(course_id):
+    """
+    This function will get all of the students enrolled in the specific course.
```

Comment:
It might be
`Returns all of the students enrolled in the specific course`

Reply:
In the `:return:` i have already told what this function is going to return.

Reply:
I think when you can tell in one line what is happening here why go for multiple lines.
Anyhow keep that one you think is more readable. :+1: :+1: :+1: